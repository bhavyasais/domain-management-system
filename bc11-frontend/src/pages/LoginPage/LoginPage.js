import React from 'react'
import { Typography, Grid, Paper, makeStyles,CircularProgress} from "@material-ui/core";
import Button from '../../components/atoms/Button/ButtonAtom'
import { useAuth0 } from "@auth0/auth0-react";
const useStyles = makeStyles((theme) => ({
    root: {
      display: 'flex',
      flexWrap: 'wrap',
      '& > *': {
        width: "500px",
        height: "300px",
        padding: "15px",
        border: '1px solid #dfe3eb',
        borderRadius: '37px',
      },
    },
  }));

function LoginPage(props){
 
    const {
        isAuthenticated,
        isLoading,
        loginWithRedirect,
        logout,
      } = useAuth0(); 

    const classes = useStyles();


    const logoutWithRedirect = () => {
        logout({
          returnTo: window.location.origin,
        });   
    } 

    return (
        <div style={{height: '98vh', width: '100%', backgroundColor: '#f8f8f9'}}>
        <Grid container alignItems="center" justify="center" style={{height: '100%', backgroundColor: 'none'}}>
            <Grid item>
            <div className={classes.root}>
                {isLoading ? <CircularProgress/>: <Paper>
                    <Grid container direction="column" alignItems="center" justify="center" style={{height: '100%', width: '100%', backgroundColor: 'none'}}>
                        {!isAuthenticated && <Grid item style={{backgroundColor: 'none', width: '100%'}}>
                            <Grid container justify="center" style={{backgroundColor: 'none', width: '100%', paddingBottom: '3%'}}>
                                <Grid item>
                                    <Typography variant="h3">
                                        Hi! Looks like you're not signed in. Would you like to sign in?
                                    </Typography>
                                </Grid>
                            </Grid>
                        </Grid>}
                        <Grid item style={{backgroundColor: 'none', width: '100%'}}>    
                            <Grid container justify="center" style={{backgroundColor: 'none', width: '100%'}}>
                                <Grid item>
                                    <Button
                                        id="signIn"
                                        variant="contained"
                                        color="primary"
                                        value={!isAuthenticated ? "Sign In" : "Log Out"}
                                        onClick={() => {!isAuthenticated ? loginWithRedirect() : logoutWithRedirect()}}
                                    />                                      
                                </Grid>
                            </Grid>                          
                        </Grid>
                    </Grid>

                </Paper>}
            </div>
            </Grid>
        </Grid>
        </div>
    )
}

export default LoginPage