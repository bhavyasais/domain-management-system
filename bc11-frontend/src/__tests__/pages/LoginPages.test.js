import React from "react";
import { act } from "react-dom/test-utils";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import LoginPage from "../../pages/LoginPage/LoginPage";

jest.mock("@auth0/auth0-react", () => ({
  useAuth0: () => ({
    isAuthenticated: false,
    loginWithRedirect: jest.fn(),
    logout: jest.fn(),
  }),
}));

describe("CollabModal", () => {
  let component;
  beforeEach(() => {
    act(() => {
      component = mount(<LoginPage />);
    });
  });
  it("should render correctly", () => {
    let wrapper;
    wrapper = renderer.create(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
  });

  it("logoutWithRedirect should be called", () => {
    expect(component).toBeDefined();
    component.find("button").at(0).props().onClick();
  });
});
