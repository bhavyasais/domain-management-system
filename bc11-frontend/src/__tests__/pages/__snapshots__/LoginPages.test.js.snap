// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollabModal should render correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#f8f8f9",
      "height": "98vh",
      "width": "100%",
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    style={
      Object {
        "backgroundColor": "none",
        "height": "100%",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div
        className="makeStyles-root-1"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
            style={
              Object {
                "backgroundColor": "none",
                "height": "100%",
                "width": "100%",
              }
            }
          >
            <div
              className="MuiGrid-root MuiGrid-item"
              style={
                Object {
                  "backgroundColor": "none",
                  "width": "100%",
                }
              }
            >
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
                style={
                  Object {
                    "backgroundColor": "none",
                    "paddingBottom": "3%",
                    "width": "100%",
                  }
                }
              >
                <div
                  className="MuiGrid-root MuiGrid-item"
                >
                  <h3
                    className="MuiTypography-root MuiTypography-h3"
                  >
                    Hi! Looks like you're not signed in. Would you like to sign in?
                  </h3>
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
              style={
                Object {
                  "backgroundColor": "none",
                  "width": "100%",
                }
              }
            >
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
                style={
                  Object {
                    "backgroundColor": "none",
                    "width": "100%",
                  }
                }
              >
                <div
                  className="MuiGrid-root MuiGrid-item"
                >
                  <div>
                    <button
                      className="MuiButtonBase-root MuiButton-root MuiButton-contained"
                      disabled={false}
                      id="signIn"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      style={
                        Object {
                          "backgroundColor": "primary",
                          "maxWidth": undefined,
                        }
                      }
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiButton-label makeStyles-buttonLabel-2"
                      >
                        <p
                          className="MuiTypography-root makeStyles-buttonLabel-2 MuiTypography-body1"
                        >
                          Sign In
                        </p>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
